python/lsst/pipe/supertask/configOverrides.py

```diff
@@ -103,3 +120,15 @@ def applyTo(self, config):
                     # this can throw
                     value = eval(value, {})
                     setattr(obj, field[-1], value)
+            elif otype == 'namesDict':
+                try:
+                    parsedNamesDict = ast.literal_eval(override)
+                    if not isinstance(parsedNamesDict, dict):
+                        raise ValueError
+                except ValueError:
+                    raise pexExceptions.RuntimeError(f"Unable parse nameTemplate {override} into a valid "
+                                                     "dict")
+                if not isinstance(config, PipelineTaskConfig):
+                    raise pexExceptions.RuntimeError("Template names can only be used on Tasks who's "
```

Comment:
`Template names` -> Dataset name substitution`.  Also `who's ConfigClass is` -> `with a ConfigClass that is`